src/Illuminate/Auth/AuthenticationException.php

```diff
@@ -3,16 +3,26 @@
 namespace Illuminate\Auth;
 
 use Exception;
+use Illuminate\Support\Collection;
 
 class AuthenticationException extends Exception
 {
+    protected $guards = [];
+
     /**
      * Create a new authentication exception.
      *
-     * @param string  $message
+     * @param string|array $guards
+     * @param string $message
      */
-    public function __construct($message = 'Unauthenticated.')
+    public function __construct($guards = [], $message = 'Unauthenticated.')
```

Comment:
Thanks @leeovery for your response. Actually I was doing exactly as you said but it gave me `[]` every time the function was executed. Here is my code:

```
    protected function unauthenticated($request, AuthenticationException $exception)
    {
        if ($request->expectsJson()) {
            return response()->json(['error' => 'Unauthenticated.'], 401);
        } else {
	        if (collect($exception->guards())->contains('admin')) {
		        return redirect()->guest(route('adminLogin'));
	        }
	        return redirect()->guest('login');
        }
    }
```
But suddenly the error has gone and it seems that everything is ok. Anyway thanks for your response